LethalAPI.Core/Patches/Fixes/FixBepInExLoggerPrefix.cs

```diff
@@ -8,161 +8,193 @@
+
+            new(OpCodes.Ret),
+
+            // new CodeInstruction(OpCodes.Ret).WithLabels(label),
+            new CodeInstruction(OpCodes.Ldarg_0).WithLabels(label),
+            new (OpCodes.Ldarg_1),
+            new (OpCodes.Ldarg_2),
+            new (OpCodes.Callvirt, Method(typeof(BepInExLogFix), nameof(ConsoleOutputLogic))),
+            new (OpCodes.Ret),
+        };
 
-        // Set default color
-        setConsoleColor.Invoke(null, new object[] { ConsoleColor.Gray });
+        for (int i = 0; i < newInstructions.Count; i++)
+            yield return newInstructions[i];
+    }
 
-        string text = (string)eventArgs.Data;
+    // ReSharper disable once RedundantAssignment
+    [HarmonyPrefix]
+    private static bool LogEventArgsPrefix(LogEventArgs __instance, ref string __result)
+    {
+        __result = __instance.Level == 0 ? __instance.Data.ToString() : GetTemplate(__instance.Level)
+            .Insert(26, __instance.Data.ToString())
+            .Insert(20, __instance.Source.SourceName)
+            .Insert(0, Log.GetDateString());
+        return false;
+    }
 
-        StringBuilder stringBuilder = new();
-        ConsoleColor oldColor = ConsoleColor.Gray;
-        bool previouslyEscaped = false;
-        for (int i = 1; i < text.Length; i++)
+    private static void ConsoleOutputLogic(ConsoleLogListener __instance, object sender, LogEventArgs eventArgs)
+    {
+        try
         {
-            char c = text[i];
-            if (c == '\\')
-            {
-                stringBuilder.Append(c);
-                previouslyEscaped = true;
-                continue;
-            }
+            // BepInEx.ConsoleManager.SetConsoleColor(Log.ColorCodes["r"]);
+            setConsoleColor.Invoke(null, new object[] { Log.ColorCodes["r"] });
 
-            if (c != '&' || previouslyEscaped)
-            {
-                previouslyEscaped = false;
-                stringBuilder.Append(c);
-                continue;
-            }
+            string text = eventArgs.ToString();
 
-            StringBuilder searchColor = new();
-            for (int j = 0; j < Log.LongestColor + 1; j++)
+            StringBuilder stringBuilder = new();
+            ConsoleColor oldColor = ConsoleColor.Gray;
+            bool previouslyEscaped = false;
+            for (int i = 0; i < text.Length; i++)
             {
-                searchColor.Append(text[i + j]);
+                char c = text[i];
+                if (c == '\\')
+                {
+                    stringBuilder.Append(c);
+                    previouslyEscaped = true;
+                    continue;
+                }
+
+                if (c != '&' || previouslyEscaped)
+                {
+                    previouslyEscaped = false;
+                    stringBuilder.Append(c);
+                    continue;
+                }
+
+                StringBuilder searchColor = new();
+                for (int j = 0; j < Log.LongestColor + 1; j++)
```

Comment:
I saw where you defined this LongestColor thing.

Why exactly do we use this, versus iterating over all colours through a non-"hardcoded" variable?

Reply:
> I saw where you defined this LongestColor thing.
> 
> Why exactly do we use this, versus iterating over all colours through a non-"hardcoded" variable?

This is because we don't want to pass the whole string through, or we may increase memory usage, and waste resources. Additionally the constant is significantly faster, and being that this executes for every single log, it can make a considerable difference from an optimization perspective. That being said I think the Log.LongestColor can probably be an auto-getter.